packages/phone-number-privacy/combiner/test/index.test.ts

```diff
@@ -181,80 +172,113 @@ describe(`POST /getContactMatches endpoint`, () => {
-      mockIsVerified.mockReturnValue(true)
-      const res = {
-        json(body: any) {
-          try {
-            expect(body.success).toEqual(true)
-            expect(body.matchedContacts).toEqual([])
-            done()
-          } catch (e) {
-            done(e)
-          }
-        },
-        status(status: any) {
-          try {
-            expect(status).toEqual(200)
-            done()
-          } catch (e) {
-            done(e)
-          }
-          return {
-            json() {
-              return {}
-            },
+      it('provides matches empty array', (done) => {
+        mockGetNumberPairContacts.mockReturnValue([])
+        mockIsVerified.mockReturnValue(true)
+        const res = {
+          json(body: any) {
+            try {
+              expect(body.success).toEqual(true)
+              expect(body.matchedContacts).toEqual([])
+              done()
+            } catch (e) {
+              done(e)
+            }
+          },
+          status(status: any) {
+            try {
+              expect(status).toEqual(200)
+              done()
+            } catch (e) {
+              done(e)
+            }
+            return {
+              json() {
+                return {}
+              },
+            }
+          },
+        } as Response
+
+        getContactMatches(req, res)
+      })
+    }
+
+    const reqs = new Map([
+      [
+        'w/o signedUserPhoneNumber',
+        {
+          body: validInput,
+          headers: mockHeaders,
+        } as Request,
+      ],
+      [
+        'w/ signedUserPhoneNumber',
+        {
+          body: {
+            ...validInput,
+            signedUserPhoneNumber: signWithDEK(validInput.userPhoneNumber, PRIVATE_KEYS[0]),
+          },
+          headers: mockHeaders,
+        } as Request,
+      ],
+    ])
+
+    reqs.forEach((req, description) =>
+      describe(description, () => {
+        expectSuccessfulMatchmaking(req)
+        describe('With replayed requests', () => {
+          mockGetDidMatchmaking.mockReturnValue(true)
+          switch (description) {
+            case 'w/o signedUserPhoneNumber':
+              it('rejects more than one request to matchmake with 403', (done) => {
+                mockGetDidMatchmaking.mockReturnValue(true)
+                mockIsVerified.mockReturnValue(true)
+                getContactMatches(req, invalidResponseExpected(done, 403))
+              })
+              break
+            case 'w/ signedUserPhoneNumber':
+              expectSuccessfulMatchmaking(req)
+            default:
+              break
           }
-        },
-      } as Response
-
-      getContactMatches(req, res)
-    })
-
-    it('rejects more than one attempt to matchmake with 403', (done) => {
-      mockGetDidMatchmaking.mockReturnValue(true)
-      mockIsVerified.mockReturnValue(true)
-      getContactMatches(req, invalidResponseExpected(done, 403))
-    })
+        })
+      })
+    )
   })
```

Comment:
It'd be also nice to test that mocked DB or blockchain failure results in success.